lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs

```diff
@@ -130,13 +131,15 @@ type Signature = Tag "signature" ByteString
 
 authorisation :: V4 -> ByteString
 authorisation V4 {..} =
-  algorithm
-    <> " Credential="
-    <> toBS metaCredential
-    <> ", SignedHeaders="
-    <> toBS metaSignedHeaders
-    <> ", Signature="
-    <> toBS metaSignature
+  mconcat
```

Comment:
*squints*

Reply:
We checked the implementation of `mconcat` for `ByteString` before making this change - it first works out exactly how much space is needed, allocates that, and then copies the bits in.

Reply:
Repeated application of `<>` leads to O(n^2) complexity - not likely to be a killer but worth getting into the habit in the system of picking the efficient alternative if there is one. It's not a big issue one way or the other, just thought I'd slip it in while trying to work on another performance issue.